docs/planning/otel/OTelWebSdk.md

```diff
@@ -0,0 +1,1344 @@
+
+- **Astro Islands**: Partial hydration with telemetry correlation across islands
+- **Qwik Resumability**: Fine-grained hydration tracking and performance monitoring
+- **Solid Start**: Streaming SSR with progressive enhancement telemetry
+- **Fresh (Deno)**: Island-based architecture with edge runtime support
+- **Enhanced SPA Frameworks**: Angular Universal, Vue Nuxt.js, React Server Components
+
+### **OTelWebSdk Capabilities & Implementation Goals**
+
+The OTelWebSdk addresses these web application challenges through a comprehensive solution that delivers:
+
+#### **Primary Goals**
+
+1. **Full OpenTelemetry API Support for Tracing and Logs**
+   - Complete OpenTelemetry compatible tracing API implementation with web-optimized performance
+   - Full OpenTelemetry logs API support for structured logging in web applications
+   - Standards compliance for interoperability with OpenTelemetry ecosystem
+   - Web-optimized implementation for performance and bundle size
+   - Browser-specific adaptations without breaking OTel contracts
+
+2. **Enterprise Multi-Tenant Support**
+   - **BOTH multi-tenant instances AND version isolation are primary goals** - not alternatives but complementary capabilities
+   - **Strategic Migration Path**: Primary goal is migrating teams from isolated versions to efficient multi-tenant instances
+   - **Version Isolation Requirement**: MUST support complete version isolation so teams can use different versions independently
+   - **Multi-version coexistence**: Different SDK versions (NPM packages or CDN) operate without conflicts within the same environment
+   - **Semantic Versioning Compatibility**: Following semver - patch/minor versions maintain backward compatibility, major versions may introduce breaking changes
+   - **Breaking Change Documentation**: All breaking changes in major versions MUST be clearly documented with migration guidance
+   - **Centralized configuration management**: Works across both multi-tenant instances and isolated versions with team-specific customization
+   - **Progressive migration support**: Teams can migrate from isolated to multi-tenant instances when ready, without coordination requirements (API surface is either identical or very similar)
+
+3. **Dynamic Configuration Management**
+   - Runtime configuration updates without application restart (following existing ApplicationInsights dynamic configuration pattern)
+   - Includes updates for connection strings, feature flags, etc.
+   - Configuration validation with rollback on failure
+   - **Remote Feature Control**: Centralized feature flag management with real-time updates
+   - **Local Override Capabilities**: Local development and testing overrides for remote configurations
+   - **Feature A/B Testing**: Built-in experimentation framework for feature rollouts and testing
+
+4. **Complete Unload and Cleanup Support**
+   - Complete removal of SDK instances with all associated hooks and timers
```

Comment:
Any special consideration about timers that need to run? What's this targeted at/do you think there are any specific engineering challenges related to this?